rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,133 @@
+
+![Figure 2](../images/0000-multi-data-collection-data-processing-images/flow-of-information.png)
+Figure 2: Flow of information in the proposed data model
+
+#### Updates to *data groups*
+Updates to *PROJECT_SUBMISSIONS* must result in updating of the associated *data groups* to trigger reprocessing. When part or the entirety of a project submission is updated the associated data group must be identified and updated with the new bundle versions if bundles have been replaced and/or with additional new bundles if bundles have been added. This is the responsibility of Ingest.  Analysis must handle the update to the *data group* by initiating only the pipelines that have been affected by the update and ensure that the output bundles of this analysis are updates to existing bundles.
+
+*Data group* updates may result from creating new bundles to add additional types of data to a project, or updating bundles to change metadata or append data to data files (e.g. FASTQ top-up).
+
+#### Implementation feasibility
+
+Both Analysis and Ingest confirm that implementation of the data_groups bundle proposed here is feasible in principle. Analysis indicates that significant changes need to be made to the existing infrastructure to handle the new type of notification. The generation of secondary data_groups of type PROJECT_SUBMISSION will be generated by analysis submitting to submission and Ingest will identify the bundles included in the secondary bundle.
+
+The following order of implementation is proposed, accounting for the dependencies between components to ensure uninterupted system functionality:
+
+1) Ingest adds functionality for supporting data_groups and starts using them for singalling submission completion
+2) Analysis adds handler for data_groups but continues to use bundle events during a transition period
+3) Analysis switches to using data_group events
+4) Ingests adds support for secondary data_groups
+5) Analyis starts notifying Ingest for the existence of data_groups
+6) Downstream components (Matrix Service, Portal) start listening for secondary data_groups
+
+### Unresolved Questions
+#### Aspects of the design to be resolved during implementation:
+* Where does the *data group* JSON schema belong?  It is not part of the experimental data model, so doesn't fit in the current hierarchy.
+
+* The bundle types for primary vs secondary *data groups* needs to be defined.
+
+#### Aspects of the design to be resolved in the future
+* *Data group* concept relates directly to the UX discussion on *experiment data sets*.  This RFC and that work could be unified with a direct mapping between *experiment* and *data groups*.
+
+## Dependencies
+
+Implementation of this RFC is partly dependent on [RFC: Bundle Types](https://github.com/HumanCellAtlas/dcp-community/pull/86), but approaches that prevent blocking are possible.
```

Comment:
It is also dependent on the analysis updates RFC

Reply:
I don't think it is blocked by the analysis updates RFC, only that some functionality using this RFC can't be implemented until the analysis updated RFC is implemented.

Reply:
True dat. Partially dependent.